src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -1132,16 +1157,60 @@ public function isKnown($option)
      * parameters should usually not be included in messages aimed at
      * non-technical people.
      *
-     * @param mixed $value The value to return the type of
+     * @param mixed  $value  The value to return the type of
+     * @param string $option The option that holds the value
      *
      * @return string The type of the value
      */
-    private function formatTypeOf($value)
+    private function formatTypeOf($value, $option)
     {
+        if (is_array($value)) {
+            foreach ($this->allowedTypes[$option] as $type) {
+                if (substr($type, -2) === '[]') {
+                    return $this->formatComplexTypeOf($value, $type);
+                }
+            }
+        }
+
         return is_object($value) ? get_class($value) : gettype($value);
     }
 
     /**
+     * Returns a string represnetation of the complex type of the value.
```

Comment:
Typo fixed, (reading through commits in your own time, or not enough work back at NG? ;-P)